pkg/agent/plugin/workloadattestor/docker/http.go

```diff
@@ -0,0 +1,31 @@
+package docker
+
+import (
+	"net/http"
+
+	"golang.org/x/time/rate"
+)
+
+// default number of requests per second
+const defaultRateLimit = 100
```

Comment:
When adding retries, my instinct is to also add rate limiting, but it may not be necessary for a local agent like the docker daemon. Also, does a rate limit actually introduce a DoS vector whereby a rogue workload could make enough attestation requests such that legitimate attestation traffic can't make it through?

Reply:
Seems like the limiting would need to be per-something rather than for the whole client.

Reply:
it could make sense to have both

limiting per container id would prevent a workload in a crash loop from DoSing us and the docker daemon by extension

limiting for the whole client helps guard against cases where workloads in a crash loop are getting new container ids as well, though maybe thats not realistic